eth: add eth_config API definitions for EIP-7910

shemnon opened the pull request:
Implements the eth_config JSON-RPC method as specified in EIP-7910 to provide comprehensive fork configuration details including current, next, and last fork configurations with their respective hashes and fork IDs.

Changes:
- Add eth_config method definition in src/eth/client.yaml
- Add ConfigurationResponse, ConfigObject, and BlobSchedule schemas
- Add bytes4 base type for CRC-32 hashes and fork IDs
- Add test cases for different scenarios

shemnon committed:
Add eth_config API definitions for EIP-7910

Implements the eth_config JSON-RPC method as specified in EIP-7910 to provide
comprehensive fork configuration details including current, next, and last
fork configurations with their respective hashes and fork IDs.

Changes:
- Add eth_config method definition in src/eth/client.yaml
- Add ConfigurationResponse, ConfigObject, and BlobSchedule schemas
- Add bytes4 base type for CRC-32 hashes and fork IDs
- Add test cases for different scenarios

shemnon committed:
Fix speling

Signed-off-by: Danno Ferrin <danno@numisight.com>

bomanaps commented on src/schemas/client.yaml:
```diff
@@ -17,3 +17,112 @@ SyncingStatus:
+      $ref: '#/components/schemas/ConfigObject'
+    nextHash:
+      title: Next configuration hash
+      description: CRC-32 hash of the the canonical JSON of the next configuration, absent if no future forks scheduled
```
```suggestion
      description: CRC-32 hash of the canonical JSON of the next configuration, absent if no future forks scheduled
```

bomanaps commented on src/schemas/client.yaml:
```diff
@@ -17,3 +17,112 @@ SyncingStatus:
+      $ref: '#/components/schemas/ConfigObject'
+    lastHash:
+      title: Last configuration hash
+      description: CRC-32 hash of the the canonical JSON of the last configured fork, absent if no future forks scheduled
```
```suggestion
      description: CRC-32 hash of the canonical JSON of the last configured fork, absent if no future forks scheduled
```

shemnon committed:
Update src/schemas/client.yaml

Co-authored-by: Mercy Boma Naps Nkari <96525594+bomanaps@users.noreply.github.com>

shemnon committed:
Update src/schemas/client.yaml

Co-authored-by: Mercy Boma Naps Nkari <96525594+bomanaps@users.noreply.github.com>

shemnon committed:
Update eth_config to match final EIP-7910 specification

- Remove configHash fields from ConfigurationResponse
- Move forkId fields into ConfigObject
- Invert precompiles map: name->address mapping with alphabetical sorting
- Update test cases to match EIP specification format

Signed-off-by: Danno Ferrin <danno@numisight.com>

shemnon committed:
Merge branch 'eip-7910-eth_config' of github.com:shemnon/ethereum-execution-apis into eip-7910-eth_config

Signed-off-by: Danno Ferrin <danno@numisight.com>

shemnon committed:
next and last must be present, and will be null if no future fork is scheduled.

Signed-off-by: Danno Ferrin <danno@numisight.com>

kclowes reviewed (approved):
lgtm!

lightclient reviewed (commented):
Generally LGTM. How were the tests generated? Would be good if we have that in rpctestgen before merging.

shemnon committed:
Merge branch 'main' into eip-7910-eth_config
